include/replayio-macros.h

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) 2024 Record Replay Inc.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// API for interacting with the record/replay driver.
+// Some parts are still in v8.h and still need to be migrated.
+
+
+#ifndef INCLUDE_REPLAYIO_MACROS_H_
+#define INCLUDE_REPLAYIO_MACROS_H_
+
+// Use this to wrap Asserts on non-trivial data, to avoid the
+// overhead of reads when Asserts are disabled.
+#define REPLAY_ASSERT(format, ...) \
+  if (recordreplay::HasAsserts()) \
+    recordreplay::Assert(format, ##__VA_ARGS__); \
```

Comment:
this would be a good place to use `AssertRaw` since we've already done the check.

Reply:
The problem was a bit tricky! I started with `AssertRaw`, but because I want to also include it in Chromium, and for some reason, we have everything doubled up between Chromium and V8, I would have to also copy-and-paste `AssertRaw` into chromium for this macro to work there, without pulling in ton of V8 dependencies as well.
I want to fix this, but first we need to move things into individual files. I have started that in the other "root context" PRs. Will come soon!